Reduce the size of the string used to name internal memory RIB tables

guvenc opened the pull request:
The allocated memory table name used by the RIB structure has a hidden limit of 25 characters in PMD. This becomes a problem when 8 digit VNIs are used and there's no NUMA(like on Bluefields) and socket_id becomes -1 and uses two characters.
So reduce the rest of the string to have enough room even with -1 in the string.

guvenc committed:
Reduce the size of the string used to name internal memory RIB tables

The allocated memory table name used by the RIB structure has a hidden limit
of 25 characters in PMD. This becomes a problem when 8 digit VNIs are
used and there's no NUMA(like on Bluefields) and socket_id becomes -1
and uses two characters.
So reduce the rest of the string to have enough room even with -1 in
the string.

Signed-off-by: Guvenc Gulce <guevenc.guelce@sap.com>

PlagueCZ reviewed (approved):
Short and simple, looks good.

guvenc committed:
Enforce 24 bit VNI range in the negative VNI test

Negative VNI test was failing for another reason and was
not testing the actual range enforcement. Correct it.

Signed-off-by: Guvenc Gulce <guevenc.guelce@sap.com>
